test/functorch/test_control_flow.py

```diff
@@ -5581,6 +5581,30 @@ def test_while_loop_schema_gen(self):
         )
         self.assertEqual(schema.parse(str(schema)), schema)
 
+    @torch._dynamo.config.patch(capture_scalar_outputs=True)
+    def test_cond_eager_run_with_item(self):
+        class M(torch.nn.Module):
+            def forward(self, a, b1, b2, c):
+                def true_fn(x):
+                    return x * b1.item()
+
+                def false_fn(x):
+                    return x * b2.item()
+
+                r = torch.cond(a, true_fn, false_fn, (c,))
+                return r * 2
```

Comment:
1) Why does this export? What does it export to?
2) Does torch.compile handle this in all situations? IIRC .item() is a graph break unless there's a config set. I don't want to get into a situation where torch.compile, torch.export, and eager-mode have different semantics

Reply:
1. After setting capture_scalar_outputs=True, we can get the following graph:
```
ExportedProgram:
    class GraphModule(torch.nn.Module):
        def forward(self, a: "b8[]", b1: "i64[1]", b2: "i64[1]", c: "f32[10]"):
             # File: /data/users/yidi/pytorch/torch/_higher_order_ops/cond.py:133 in cond, code: return cond_op(pred, true_fn, false_fn, operands)
            true_graph_0 = self.true_graph_0
            false_graph_0 = self.false_graph_0
            cond = torch.ops.higher_order.cond(a, true_graph_0, false_graph_0, [c, b1, b2]);  a = true_graph_0 = false_graph_0 = c = b1 = b2 = None
            getitem: "f32[10]" = cond[0];  cond = None
            
             # File: /data/users/yidi/pytorch/test/functorch/test_control_flow.py:5595 in forward, code: return r * 2
            mul: "f32[10]" = torch.ops.aten.mul.Tensor(getitem, 2);  getitem = None
            return (mul,)
            
        class true_graph_0(torch.nn.Module):
            def forward(self, c: "f32[10]", b1: "i64[1]", b2: "i64[1]"):
                 # File: /data/users/yidi/pytorch/test/functorch/test_control_flow.py:5589 in true_fn, code: return x * b1.item()
                item: "Sym(u8)" = torch.ops.aten.item.default(b1);  b1 = None
                mul: "f32[10]" = torch.ops.aten.mul.Tensor(c, item);  c = item = None
                return (mul,)
                
        class false_graph_0(torch.nn.Module):
            def forward(self, c: "f32[10]", b1: "i64[1]", b2: "i64[1]"):
                 # File: /data/users/yidi/pytorch/test/functorch/test_control_flow.py:5592 in false_fn, code: return x * b2.item()
                item: "Sym(u9)" = torch.ops.aten.item.default(b2);  b2 = None
                mul: "f32[10]" = torch.ops.aten.mul.Tensor(c, item);  c = item = None
                return (mul,)
```
3. Yeah, all mode need to patch with @torch._dynamo.config.patch(capture_scalar_outputs=True) for control flow operators since we turned on dynamo all the time.